msal/tests/Test.MSAL.NET.Integration/UnifiedCacheTests.cs

```diff
@@ -0,0 +1,491 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+extern alias msal;
+extern alias adal;
+
+using Test.Microsoft.Identity.LabInfrastructure;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Net;
+using System.Security;
+using System.Threading.Tasks;
+using Test.MSAL.NET.Integration.Infrastructure;
+using System.IO;
+using System.Linq;
+using System;
+
+namespace Test.MSAL.NET.Integration
+{
+    [TestClass]
+    public class UnifiedCacheTests
```

Comment:
>UnifiedCacheTests [](start = 17, length = 17)

These tests are not able to run in parallel. I'm thinking it should be possible to structure the tests in a way which doesn't prevent them from running in parallel.

Reply:
ok, i will make in memory persisted cache instance specific so individual tests will be isolated.

Reply:
any update on this change?

---
In reply to: [223480524](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1304#discussion_r223480524) [](ancestors = 223480524)

Reply:
yes it was commited